crates/astria-sequencer/src/app/mod.rs

```diff
@@ -978,35 +1125,23 @@ impl App {
         // get deposits for this block from state's ephemeral cache and put them to storage.
         let mut state_tx = StateDelta::new(self.state.clone());
         let deposits_in_this_block = self.state.get_cached_block_deposits();
-        debug!(
-            deposits = %telemetry::display::json(&deposits_in_this_block),
-            "got block deposits from state"
-        );
+        debug!(deposits = %json(&deposits_in_this_block), "got block deposits from state");
 
         state_tx
             .put_deposits(&block_hash, deposits_in_this_block.clone())
             .wrap_err("failed to put deposits to state")?;
 
-        let vote_extensions_enable_height = self
-            .state
-            .get_vote_extensions_enable_height()
-            .await
-            .wrap_err("failed to get vote extensions enabled height")?;
-        let vote_extensions_enabled = vote_extensions_enable_height <= height.value();
-        let injected_txs_count = if vote_extensions_enabled {
-            INJECTED_TRANSACTIONS_COUNT_AFTER_VOTE_EXTENSIONS_ENABLED
-        } else {
-            INJECTED_TRANSACTIONS_COUNT_BEFORE_VOTE_EXTENSIONS_ENABLED
-        };
-
         // cometbft expects a result for every tx in the block, so we need to return a
-        // tx result for the commitments, even though they're not actually user txs.
+        // tx result for the commitments and other injected data items, even though they're not
+        // actually user txs.
         //
         // the tx_results passed to this function only contain results for every user
-        // transaction, not the commitment, so its length is len(txs) - 3.
-        let mut finalize_block_tx_results: Vec<ExecTxResult> = Vec::with_capacity(txs.len());
+        // transaction, not the injected ones.
+        let non_rollup_tx_count = parsed_data_items.non_rollup_transaction_count();
```

Comment:
rename to `non_injected_tx_count`? the naming here is a bit confusing, as `Transfer` is not a rollup tx for example

Reply:
Good point.  I see you mentioned this in a few places.  I'll aim to use "user submitted tx" where I was using "rollup tx" and "injected tx" for the others.

Reply:
Done in [4a045d3](https://github.com/astriaorg/astria/pull/1917/commits/4a045d36c177676c3af883aed37da3cce61f30b6).